src/execution_impl.rs

```diff
@@ -83,21 +83,16 @@ pub(crate) fn exec(
     param: &str,
     interface: &dyn Interface,
 ) -> Result<Response> {
+    let mut env = Env::new(interface);
     let instance = match instance {
         Some(instance) => instance,
-        None => create_instance(limit, module, interface)?,
+        None => create_instance(limit, module, env.clone())?,
     };
-    let mut env = as_ffi_bindings::Env::default();
```

Comment:
I'm not sure why you previously would create another env here.

Id' say keep the creating of the `Env` inside `create_instance`, and simply return from there alongside the instance. 

If you need this env pointer below, I'd say use `Env.wasm_env`, from the env created inside `create_instance`.